src/transactions/test/SorobanTxTestUtils.cpp

```diff
@@ -271,6 +271,96 @@ makeTransferEvent(SCAddress const& from, SCAddress const& to, int64_t amount,
     return ContractEvent();
 }
 
+ContractEvent
+makeTransferEvent(const stellar::Hash& contractId, Asset const& asset,
```

Comment:
nit: I think this function has too complex logic for a test helper. I think it would be cleaner to pass std::optional<SCVal> for toMuxId, so there is no need for duplicating the memo mapping logic here (generally, the more declarative tests are, the better).

Reply:
yeah good point. I've refactor this to pass in optional toMuxID entry directly instead of constructing it inside the helper, that should simplify it a bit.